### ISSUE TITLE
Kibana chart doesn't support Elasticsearch authentication

### ISSUE THREAD

**Issue by rekibnikufesin** [bitnami/charts#7450]
**Which chart**:
bitnami/kibana 8.2.3

**Describe the bug**
Elasticsearch authentication credentials are not written to `kibana.yml` file. As a result, kibana is unable to launch and configure the indices necessary for installation/operation

**To Reproduce**

with the following as `values.yaml`:
```
elasticsearch:
  hosts:
    - elasticsearch-coordinating-only
  port: 9200
  security:
    auth:
      enabled: true
      kibanaUsername: "my-username"
      existingSecret: kibana-passwd
```
install kibana using the helm chart: `helm install kibana bitnami/kibana --version 8.2.3 -f values.yaml`

**Expected behavior**
Credentials should be written to `/opt/bitnami/kibana/config/kibana.yml`. Instead, no credentials are written to `kibana.yml`. Logs show the following (note the `anonymous_user`):
```
{"type":"log","@timestamp":"2021-09-09T21:47:16+00:00","tags":["error","savedobjects-service"],"pid":53,"message":"[.kibana_task_manager] Action failed with 'security_exception: [security_exception] Reason: action [indices:admin/get] is unauthorized for user [anonymous_user] with roles [monitor_role], this action is granted by the index privileges [view_index_metadata,manage,all]'. Retrying attempt 7 in 64 seconds."}
```

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.0", GitCommit:"7f2df6467771a75f5646b7f12afb408590ed1755", GitTreeState:"dirty", GoVersion:"go1.16.4"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:59:11Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:02:01Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
This problem exists regardless of using `existingSecret` or `kibanaPassword`

**Comment by yilmi** [comment]
Hi @rekibnikufesin! 

The password is set through the `KIBANA_PASSWORD` environment variable which you can see [here](https://github.com/bitnami/charts/blob/master/bitnami/kibana/templates/deployment.yaml#L115). The container then uses it to set the password [libkibana.sh](https://github.com/bitnami/bitnami-docker-kibana/blob/f68ff875006f49cbb4b5901c2dae2b3e64ff2619/7.10.2/debian-10/rootfs/opt/bitnami/scripts/libkibana.sh#L119-L120).

Now this happens only if the configuration file is writeable, could you check if you have [this message](https://github.com/bitnami/bitnami-docker-kibana/blob/f68ff875006f49cbb4b5901c2dae2b3e64ff2619/7.10.2/debian-10/rootfs/opt/bitnami/scripts/libkibana.sh#L46) in your logs?
>The Kibana configuration file '${KIBANA_CONF_FILE}' is not writable. Configurations based on environment variables will not be applied for this file."

You could also check if you have any volumes left from the install and delete them to make sure they get created upon install.

**Comment by rekibnikufesin** [comment]
Hey @yilmi!
Thanks for the feedback.
I don't see that message in the logs anywhere.

I also don't see the environment variable being set. Here are the environment variables in the container:
```
I have no name!@kibana-856d9dddd-pnjd6:/$ env | grep KIBANA
KIBANA_SERVICE_HOST=10.107.108.145
KIBANA_PORT_5601_TCP=tcp://10.107.108.145:5601
KIBANA_SERVICE_PORT_HTTP=5601
KIBANA_PORT_5601_TCP_PORT=5601
KIBANA_SERVICE_PORT=5601
KIBANA_PORT_NUMBER=5601
KIBANA_PORT_5601_TCP_PROTO=tcp
KIBANA_FORCE_INITSCRIPTS=false
KIBANA_PORT_5601_TCP_ADDR=10.107.108.145
KIBANA_ELASTICSEARCH_PORT_NUMBER=9200
KIBANA_PORT=tcp://10.107.108.145:5601
KIBANA_ELASTICSEARCH_URL=http://elasticsearch-coordinating-only.kamino-common:9200
```

I tried running the `libkibana.sh` script manually. It failed with `permission denied`.
```
I have no name!@kibana-5ccbc5b8fb-cz9th:/$ /opt/bitnami/scripts/libkibana.sh
bash: /opt/bitnami/scripts/libkibana.sh: Permission denied
```

It looks like it may be a permissions issue with the `securityContext`. I tried using `securityContext.enabled: false` and got the same results.
I'm going to try again using UID 0 as the `runAsUser`

**Comment by rekibnikufesin** [comment]
Trying as UID 0 produced the same results, but #7412 is fixed and merged so I updated to chart version 9.0.2 and it works as expected.
Thanks for your help!
